pkg/services/validator/aws.go

```diff
@@ -57,44 +68,421 @@ func readAwsPlugin(vc *components.ValidatorConfig, k8sClient kubernetes.Interfac
+		r := r
+		if err := readIamGroupRule(c, &r, i, ruleNames); err != nil {
+			return err
+		}
+	}
+	addRules := true
+	if c.Validator.IamGroupRules == nil {
+		c.Validator.IamGroupRules = make([]vpawsapi.IamGroupRule, 0)
+	} else {
+		addRules, err = prompts.ReadBool("Add another group rule", false)
+		if err != nil {
+			return err
+		}
+	}
+	if !addRules {
+		return nil
+	}
+	for {
+		if err := readIamGroupRule(c, nil, -1, ruleNames); err != nil {
+			return err
+		}
+		add, err := prompts.ReadBool("Add another group rule", false)
+		if err != nil {
+			return err
+		}
+		if !add {
+			break
+		}
+	}
+	return nil
+}
+
+func readIamGroupRule(c *components.AWSPluginConfig, r *vpawsapi.IamGroupRule, idx int, ruleNames *[]string) error {
+	if r == nil {
+		r = &vpawsapi.IamGroupRule{
+			IamGroupName: "",
+			Policies:     []vpawsapi.PolicyDocument{},
+		}
+	}
+	err := initAwsRule(r, "iam group", ruleNames)
+	if err != nil {
+		return err
+	}
+	if r.IamGroupName == "" {
+		groupName, err := prompts.ReadText("IAM Group Name", "", false, -1)
+		if err != nil {
+			return err
+		}
+		r.IamGroupName = groupName
+	}
+
+	addPolicies := true
+	for addPolicies {
+		policyDoc, err := readIamPolicy()
+		if err != nil {
+			return err
+		}
+
+		r.Policies = append(r.Policies, policyDoc)
+		addPolicies, err = prompts.ReadBool("Add another policy document", false)
+		if err != nil {
+			return err
+		}
+	}
+	if idx == -1 {
+		c.Validator.IamGroupRules = append(c.Validator.IamGroupRules, *r)
+	} else {
+		c.Validator.IamGroupRules[idx] = *r
+	}
+	return nil
+}
+
+func configureIamPolicyRules(c *components.AWSPluginConfig, ruleNames *[]string) error {
+	log.InfoCLI(`
+	AWS IAM Policy validation ensures that specified IAM Policies have every permission
+	specified in the passed in policy documents.
+	`)
+
+	validatePolicies, err := prompts.ReadBool("Enable Iam Policy validation", true)
+	if err != nil {
+		return err
+	}
+	if !validatePolicies {
+		c.Validator.IamPolicyRules = nil
+		return nil
+	}
+	for i, r := range c.Validator.IamPolicyRules {
+		r := r
+		if err := readIamPolicyRule(c, &r, i, ruleNames); err != nil {
+			return err
+		}
+	}
+	addRules := true
+	if c.Validator.IamPolicyRules == nil {
+		c.Validator.IamPolicyRules = make([]vpawsapi.IamPolicyRule, 0)
+	} else {
+		addRules, err = prompts.ReadBool("Add another policy rule", false)
+		if err != nil {
+			return err
+		}
+	}
+	if !addRules {
+		return nil
+	}
+	for {
+		if err := readIamPolicyRule(c, nil, -1, ruleNames); err != nil {
+			return err
+		}
+		add, err := prompts.ReadBool("Add another policy rule", false)
+		if err != nil {
+			return err
+		}
+		if !add {
+			break
+		}
+	}
+	return nil
+}
+
+func readIamPolicyRule(c *components.AWSPluginConfig, r *vpawsapi.IamPolicyRule, idx int, ruleNames *[]string) error {
+	if r == nil {
+		r = &vpawsapi.IamPolicyRule{
+			IamPolicyARN: "",
+			Policies:     []vpawsapi.PolicyDocument{},
+		}
+	}
+	err := initAwsRule(r, "iam policy", ruleNames)
```

Comment:
nit: please use all caps IAM here and in the 3 other spots where you're calling `initAwsRule`